### DISCUSSION TITLE
New api documentation not clear for mutationStore

### DISCUSSION THREAD

**Discussion by lionelB** [urql-graphql/urql#2746] [Q&A] [answered]
I'm a bit lost with the new API using mutationStore.
In the previous API (before v2) we use a `mutation(operationStore)` that return a promise

In the new API, the docs say that mutationStore `returns the OperationResult as a store`. 
It seems not very practicable to use a store + subscribe to handle an async call (when we potentially have a first call with fetching=true, and a second with a fetching false + error or data) 

Refering the documentation (https://formidable.com/open-source/urql/docs/basics/svelte/#handling-mutation-errors)
> It's worth noting that the promise we receive when calling the execute function will never reject. Instead it will always return a promise that resolves to an mutationStore, even if the mutation has failed.

I can't find an example with a promise (the example below doesn't explain how the `mutateTodo` was created 

May have miss something ?

**Comment by kitten** [comment, marked as answer]
If you just want a promisified call you can just use the core `Client` directly without the bindings: https://formidable.com/open-source/urql/docs/basics/core/#one-off-queries-and-mutations

This wasn't really clear in the last version before, due to Svelte sometimes having a bit of an API design conflict between reactive stores and promisified results.

Ultimately, the store now tells you whether something is in progress or not and your UI can react accordingly, just like with other bindings, and just like with queries.

This is more of a question of course of how you want to handle mutations, since with those you only care about one result. But if you're after a promise then you don't technically would have to use the Svelte-specific bindings for your mutation

**Reply by lionelB** [reply]
thanks for the reply :+1: 
I probably also need to change a bit my previous habits switching from promise to store
